src/qt/pivx/masternodeswidget.cpp

```diff
@@ -197,29 +205,91 @@ void MasterNodesWidget::onEditMNClicked(){
     }
 }
 
-void MasterNodesWidget::startAlias(QString strAlias){
+void MasterNodesWidget::startAlias(QString strAlias) {
     QString strStatusHtml;
     strStatusHtml += "Alias: " + strAlias + " ";
 
     for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
         if (mne.getAlias() == strAlias.toStdString()) {
             std::string strError;
-            CMasternodeBroadcast mnb;
-            if (CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb)) {
-                strStatusHtml += "successfully started.";
-                mnodeman.UpdateMasternodeList(mnb);
-                mnb.Relay();
-                mnModel->updateMNList();
-            } else {
-                strStatusHtml += "failed to start.\nError: " + QString::fromStdString(strError);
-            }
+            strStatusHtml += (!startMN(mne, strError)) ? ("failed to start.\nError: " + QString::fromStdString(strError)) : "successfully started.";
             break;
         }
     }
-    inform(strStatusHtml);
+    // update UI and notify
+    updateModelAndInform(strStatusHtml);
+}
+
+void MasterNodesWidget::updateModelAndInform(QString informText) {
+    mnModel->updateMNList();
+    inform(informText);
+}
+
+bool MasterNodesWidget::startMN(CMasternodeConfig::CMasternodeEntry mne, std::string& strError) {
+    CMasternodeBroadcast mnb;
+    if (!CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb))
+        return false;
+
+    mnodeman.UpdateMasternodeList(mnb);
+    mnb.Relay();
+    return true;
+}
+
+void MasterNodesWidget::onStartAllClicked(int type) {
+    if(!verifyWalletUnlocked()) return;
+    if (isLoading) {
+        inform(tr("Background task is being executed, please wait"));
+    } else {
+        isLoading = true;
+        if (!execute(type)) {
+            isLoading = false;
+            inform(tr("Cannot perform Mastenodes start"));
+        }
+    }
+}
+
+bool MasterNodesWidget::startAll(QString& failText, bool onlyMissing) {
+    int amountOfMnFailed = 0;
+    int amountOfMnStarted = 0;
+    for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
+        // Check for only missing
+        if (onlyMissing && !mnModel->isMNMissingOrExpired(QString::fromStdString(mne.getAlias())))
```

Comment:
I think we should still provide feedback to the user here, indicating that not all masternodes have been started (i.e. `amountOfMnFailed++` before `continue`).
Except when the status is ENABLED.

Reply:
Missing is different than collateral vin spent or any other error status.
But well, not going to enter in semantics.

Pushed.